gfootball/env/observation_processor.py

```diff
@@ -138,6 +171,34 @@ def softmax(x):
   return np.exp(x) / np.sum(np.exp(x), axis=0)
 
 
+def pprint_players_actions(writer, players_actions):
+    table_text = [["TEAM", "PLAYER", "SPRINT", "DRIBBLE", "DIRECTION",
+                   "ACTION"]]
+    widths = [35, 50, 50, 55, 60, 50]
+
+    team_short_name = {'left': 'L',
+                       'right': 'R'}
+    direction_short_name = {'-': '-',
+                            'top': 'TT',
+                            'top_right': 'TR',
+                            'right': 'RR',
+                            'bottom_right': 'BR',
+                            'bottom': 'BB',
+                            'bottom_left': 'BL',
+                            'left': 'LL',
+                            'top_left': 'TL'}
+
+    for team in ('left', 'right'):
+        for _, player_actions in sorted(players_actions[team].items()):
```

Comment:
Table is sorted by player number, right? If one controls 2 players on the left team order of players in the table depends on the specific players being controlled at a given time. I think order should be stable - ie. 1st entry always represents the 1st player from the observations, so that it is controlled by the same agent